File: timeConversion/code.rb
#!/bin/ruby

require 'json'
require 'stringio'

#
# Complete the 'timeConversion' function below.
#
# The function is expected to return a STRING.
# The function accepts STRING s as parameter.
#

def timeConversion(s)
    # Write your code here
    time_arr = s.split(":")
    hours = time_arr[0].to_i
    minutes = time_arr[1]
    seconds = time_arr[2]
    meridian = s[-2..-1]
    if meridian == "PM"
        hours += 12 if hours != 12
        seconds.slice! "PM"
    else
        if hours == 12
            hours = "00"
        elsif hours < 10
            hours = "0#{hours}"
        end
        seconds.slice! "AM"
    end
    "#{hours}:#{minutes}:#{seconds}"

end

fptr = File.open(ENV['OUTPUT_PATH'], 'w')

s = gets.chomp

result = timeConversion s

fptr.write result
fptr.write "\n"

fptr.close()

File: DivisiblePair/code.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'divisibleSumPairs' function below.
#
# The function is expected to return an INTEGER.
# The function accepts following parameters:
#  1. INTEGER n
#  2. INTEGER k
#  3. INTEGER_ARRAY ar
#

def divisibleSumPairs(n, k, ar):
    # Write your code here
    #!/bin/ruby

require 'json'
require 'stringio'

#
# Complete the 'divisibleSumPairs' function below.
#
# The function is expected to return an INTEGER.
# The function accepts following parameters:
#  1. INTEGER n
#  2. INTEGER k
#  3. INTEGER_ARRAY ar
#

def divisibleSumPairs(n, k, ar):
    # Write your code here
    #look to the first element then find the second that match
    # element=k-arr[i]
    pairs = 0
    i = 0
    while i < ar.length - 1:
        j = i + 1
        while j < ar.length:
            if (ar[i] + ar[j]) % k == 0:
                pairs += 1
            j += 1
        i += 1
    pairs


fptr = File.open(ENV['OUTPUT_PATH'], 'w')

first_multiple_input = gets.rstrip.split

n = first_multiple_input[0].to_i

k = first_multiple_input[1].to_i

ar = gets.rstrip.split.map(&:to_i)

result = divisibleSumPairs n, k, ar

fptr.write result
fptr.write "\n"

fptr.close()


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    first_multiple_input = input().rstrip().split()

    n = int(first_multiple_input[0])

    k = int(first_multiple_input[1])

    ar = list(map(int, input().rstrip().split()))

    result = divisibleSumPairs(n, k, ar)

    fptr.write(str(result) + '\n')

    fptr.close()
